packages/netlify/src/lib/nft.ts

```diff
@@ -0,0 +1,85 @@
+import { relative as relativePath } from 'node:path';
+import { fileURLToPath } from 'node:url';
+import { copyFilesToFolder } from '@astrojs/internal-helpers/fs';
+import type { AstroIntegrationLogger } from 'astro';
+
+// Based on the equivalent function in `@astrojs/vercel`
+export async function copyDependenciesToFunction(
+	{
+		entry,
+		outDir,
+		includeFiles,
+		excludeFiles,
+		logger,
+	}: {
+		entry: URL;
+		outDir: URL;
+		includeFiles: URL[];
+		excludeFiles: URL[];
+		logger: AstroIntegrationLogger;
+	},
+	// we want to pass the caching by reference, and not by value
+	cache: object
+): Promise<{ handler: string }> {
+	const entryPath = fileURLToPath(entry);
+	logger.info(`Bundling function ${relativePath(fileURLToPath(outDir), entryPath)}`);
+
+	// Get root of folder of the system (like C:\ on Windows or / on Linux)
+	let base = entry;
+	while (fileURLToPath(base) !== fileURLToPath(new URL('../', base))) {
+		base = new URL('../', base);
+	}
+
+	// The Vite bundle includes an import to `@vercel/nft` for some reason,
+	// and that trips up `@vercel/nft` itself during the adapter build. Using a
+	// dynamic import helps prevent the issue.
+	// TODO: investigate why
+	const { nodeFileTrace } = await import('@vercel/nft');
+	const result = await nodeFileTrace([entryPath], {
+		base: fileURLToPath(base),
+		// If you have a route of /dev this appears in source and NFT will try to
+		// scan your local /dev :8
+		ignore: ['/dev/**'],
+		cache,
+	});
+
+	for (const error of result.warnings) {
+		if (error.message.startsWith('Failed to resolve dependency')) {
+			const [, module, file] =
+				/Cannot find module '(.+?)' loaded from (.+)/.exec(error.message) || [];
+
+			// The import(astroRemark) sometimes fails to resolve, but it's not a problem
+			if (module === '@astrojs/') continue;
+
+			// Sharp is always external and won't be able to be resolved, but that's also not a problem
+			if (module === 'sharp') continue;
+
+			if (entryPath === file) {
+				logger.debug(
+					`[@astrojs/netlify] The module "${module}" couldn't be resolved. This may not be a problem, but it's worth checking.`
+				);
+			} else {
+				logger.debug(
+					`[@astrojs/netlify] The module "${module}" inside the file "${file}" couldn't be resolved. This may not be a problem, but it's worth checking.`
```

Comment:
`[@astrojs/netlify]` isn't needed. The logger provided to the adapter is already scoped to prefix `@astrojsd/netlify` to each log